rotkehlchen/history/events.py

```diff
@@ -355,34 +358,35 @@ def new_step_cb(state_name: str) -> None:
         step = self._increase_progress(step, total_steps)
 
         self.processing_state_name = 'Querying ethereum transactions history'
-        ethereum = self.chains_aggregator.get_chain_manager(SupportedBlockchain.ETHEREUM)
-        tx_filter_query = EvmTransactionsFilterQuery.make(
-            limit=None,
-            offset=None,
-            addresses=None,
-            # We need to have history of transactions since before the range
-            from_ts=Timestamp(0),
-            to_ts=end_ts,
-            chain_id=ChainID.ETHEREUM,
-        )
-        try:
-            ethereum.transactions.query_chain(filter_query=tx_filter_query)
-        except RemoteError as e:
-            msg = str(e)
-            self.msg_aggregator.add_error(
-                f'There was an error when querying etherscan for ethereum transactions: {msg}'
-                f'The final history result will not include ethereum transactions',
+        for blockchain in (SupportedBlockchain.ETHEREUM, SupportedBlockchain.OPTIMISM):
+            evm_manager = self.chains_aggregator.get_chain_manager(blockchain)  # type: ignore[arg-type]  # noqa: E501
```

Comment:
what's the type error here for?

Reply:
It was because mypy was not figuring out that `blockchain` is not just any `SupportedBlcokchain`, but is limited to `SupportedBlockchain.ETHEREUM` and `SupportedBlockchain.OPTIMISM`. After adding strict typing to `EVM_CHAINS_WITH_TRANSACTIONS` constant it now works fine and I removed the `type: ignore`